easier68k/core/opcodes/eor.py

```diff
@@ -0,0 +1,278 @@
+from ..models.memory_value import MemoryValue
+
+
+class Eor(Opcode):  # Forward declaration
+    pass
+
+
+class Eor(Opcode):
+    valid_sizes = [OpSize.BYTE, OpSize.WORD, OpSize.LONG]
+
+    def __init__(self, params: list, size: OpSize=OpSize.WORD):
+        assert len(params) == 2
+        assert isinstance(params[0], AssemblyParameter)
+        assert isinstance(params[1], AssemblyParameter)
+
+        # check src param is valid
+        assert params[0].mode == EAMode.DRD
+        self.src = params[0]
+
+        # check the dest param is valid
+        assert params[1].mode != EAMode.ARD or params[1].mode != EAMode.IMM
+        self.dest = params[1]
+
+        assert size in Eor.valid_sizes
+        self.size = size
+
+    def assemble(self) -> bytearray:
+        """
+        Assembles this opcode into hex to be inserted into memory
+        :return: The hex version of this opcode
+        """
+        # create the opcode
+        ret_opcode = 0b1011 << 12
+        ret_opcode |= self.src.data << 9
+
+        if self.size == OpSize.BYTE:
+            ret_opcode |= 0b100 << 6
+        elif self.size == OpSize.WORD:
+            ret_opcode |= 0b101 << 6
+        elif self.size == OpSize.LONG:
+            ret_opcode |= 0b110 << 6
+
+        ret_opcode |= ea_mode_bin.parse_from_ea_mode_modefirst(self.dest) << 0
+
+        ret_bytes = bytearray(ret_opcode.to_bytes(OpSize.WORD.value, byteorder='big', signed=False))
+
+        if self.dest.mode == EAMode.AWA or self.dest.mode == EAMode.ALA:
+            ret_bytes.extend(opcode_util.ea_to_binary_post_op(self.dest, self.size).get_value_bytearray())
+
+        return ret_bytes
+
+    def execute(self, simulator: M68K):
+        """
+        Executes this command in a simulator
+        :param simulator: The simulator to execute the command on
+        :return: Nothing
+        """
+        # get the length
+        val_length = self.size.get_number_of_bytes()
+
+        # get the value of src from the simulator
+        src_val = self.src.get_value(simulator, val_length)
+
+        # get the value of dest from the simulator
+        dest_val = self.dest.get_value(simulator, val_length)
+
+        # increment the program counter by the length of the instruction (1 word)
+        to_increment = OpSize.WORD.value
+
+        # repeat for the dest
+        if self.dest.mode in [EAMode.AbsoluteLongAddress]:
+            to_increment += OpSize.LONG.value
+
+        if self.dest.mode in [EAMode.AbsoluteWordAddress]:
+            to_increment += OpSize.WORD.value
+
+        result_unsigned = src_val.get_value_unsigned() ^ dest_val.get_value_unsigned()
+
+        msb_bit = 0
+
+        if self.size is OpSize.BYTE:
+            msb_bit = 0x80
+        elif self.size is OpSize.WORD:
+            msb_bit = 0x8000
+        elif self.size is OpSize.LONG:
+            msb_bit = 0x80000000
+
+        simulator.set_condition_status_code(ConditionStatusCode.N, msb_bit & result_unsigned != 0)
```

Comment:
Easy68k uses a global function for setting the CCR bits that, for the most part, is fairly consistent. Given your experience with the logic surrounding the CCR bits, do you think that there are any patterns that exist that could allow for more code reuse?

Reply:
Hmm, it does seem possible to have a global function that can take in an OPCODE (with some extra parameters) and change the CCR accordingly, since the concept for each CCR bit is pretty consistent for most of the OPCODEs. This [tutorial](http://mrjester.hapisan.com/04_MC68/Sect06Part01/Index.html) goes more in-depth as to how they are affected.

If we don't want to go the global function route, I have noticed that all OPCODE derivatives affect the CCR in the same way as the base OPCODE. For example, SUBI and SUBQ all affect the CCR in the same way as SUB, their base OPCODE. This seems consistent for all OPCODEs and their derivatives, so we could design the derivative OPCODEs to inherit from the base OPCODEs to have the same implementation for the CCR.

Reply:
I think that a combination of these approaches would be something useful to consider down the line.